network/p2p/middleware.go

```diff
@@ -524,18 +568,32 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 		m.wg.Add(1)
 		go func(msg *message.Message) {
 			defer m.wg.Done()
-			if decodedMsgPayload, err := m.codec.Decode(msg.Payload); err != nil {
+			decodedMsgPayload, err := m.codec.Decode(msg.Payload)
+			if err != nil {
 				m.log.
 					Warn().
 					Err(fmt.Errorf("could not decode message: %w", err)).
 					Hex("sender", msg.OriginID).
 					Hex("event_id", msg.EventID).
 					Str("event_type", msg.Type).
 					Str("channel", msg.ChannelID)
+				return
+			}
+
+			channel := channels.Channel(msg.ChannelID)
+			if err := m.validateUnicastAuthorizedSender(ctx, remotePeer, channel, decodedMsgPayload); err != nil {
+				m.log.
+					Error().
+					Err(err).
+					Hex("sender", msg.OriginID).
+					Hex("event_id", msg.EventID).
+					Str("event_type", msg.Type).
+					Str("channel", msg.ChannelID).
+					Msg("unicast authorized sender validation failed")
 			} else {
```

Comment:
nit: make this consistent with the previous conditional and return after the authorization error
```suggestion
				return
			}
```

Reply:
https://github.com/onflow/flow-go/pull/2832/commits/9a988aed31fd07ef859b72aff9f2522af447dc9c